lib/route/addr.c

```diff
@@ -840,6 +851,25 @@ int rtnl_addr_delete(struct nl_sock *sk, struct rtnl_addr *addr, int flags)
  * @{
  */
 
+int rtnl_addr_set_alias(struct rtnl_addr *addr, const char *alias)
+{
+	if (strlen(alias) > sizeof(addr->a_alias) - 1)
+		return -NLE_RANGE;
+
+	strcpy(addr->a_alias, alias);
```

Comment:
maybe use `_nl_strncpy_assert()`, because that takes care to zero out the trailing bytes.